Find a functionally equivalent code:android.net.Proxy.validate:COMMENT
Method Modifier: public      static      hidden      
Comment:/**
 * Validate syntax of hostname, port and exclusion list entries
 * {@hide}
 */

@@ -1,27 +1,23 @@
 {
     Matcher match = HOSTNAME_PATTERN.matcher(hostname);
     Matcher listMatch = EXCLLIST_PATTERN.matcher(exclList);
-    if (!match.matches()) {
-        throw new IllegalArgumentException();
-    }
-    if (!listMatch.matches()) {
-        throw new IllegalArgumentException();
-    }
-    if (hostname.length() > 0 && port.length() == 0) {
-        throw new IllegalArgumentException();
-    }
+    if (!match.matches())
+        return PROXY_HOSTNAME_INVALID;
+    if (!listMatch.matches())
+        return PROXY_EXCLLIST_INVALID;
+    if (hostname.length() > 0 && port.length() == 0)
+        return PROXY_PORT_EMPTY;
     if (port.length() > 0) {
-        if (hostname.length() == 0) {
-            throw new IllegalArgumentException();
-        }
+        if (hostname.length() == 0)
+            return PROXY_HOSTNAME_EMPTY;
         int portVal = -1;
         try {
             portVal = Integer.parseInt(port);
         } catch (NumberFormatException ex) {
-            throw new IllegalArgumentException();
-        }
-        if (portVal <= 0 || portVal > 0xFFFF) {
-            throw new IllegalArgumentException();
+            return PROXY_PORT_INVALID;
         }
+        if (portVal <= 0 || portVal > 0xFFFF)
+            return PROXY_PORT_INVALID;
     }
+    return PROXY_VALID;
 }

